src/libs/ajax.js

```diff
@@ -58,10 +74,14 @@ const query = (url, data, method = 'GET', options = {}) => {
     };
     if (method === 'GET') {
       xhr.open(method, `${url}?${dataToUrl(data)}`);
+      setHeaders(xhr, headers);
       xhr.send();
     } else {
       xhr.open(method, url);
-      xhr.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+      setHeaders(xhr, headers);
+      if (!headersContain(headers, 'Content-Type')) {
+        xhr.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
```

Comment:
shouldn't this header be set on POST requests only ?

Reply:
I didn't change the method check. Do you want me to check for `POST` only?

Reply:
Yes. Before this change we used only `GET` and `POST`, and this header should be defined only when JSON data is present in body.